python/lsst/cp/pipe/measureCrosstalk.py

```diff
@@ -0,0 +1,799 @@
+            and source chip, then by target and source amplifier.
+        inputFluxes : `list` [`dict` [`dict` [`list`]]]
+            A list of nested dictionaries of source pixel fluxes, indexed
+            by source chip and amplifier.
+        camera : `lsst.afw.cameraGeom.Camera`
+            Input camera.
+        inputDims : `list` [`lsst.daf.butler.DataCoordinate`]
+            DataIds to use to construct provenance.
+        outputDims : `list` [`lsst.daf.butler.DataCoordinate`]
+            DataIds to use to populate the output calibration.
+
+        Returns
+        -------
+        results : `lsst.pipe.base.Struct`
+            The results struct containing:
+
+            ``outputCrosstalk`` : `lsst.ip.isr.CrosstalkCalib`
+                Final crosstalk calibration.
+            ``outputProvenance`` : `lsst.ip.isr.IsrProvenance`
+                Provenance data for the new calibration.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if the input data contains too many target
+            detectors.
+
+        Notes
+        -----
+        The lsstDebug.Info() method can be rewritten for __name__ =
+        `lsst.ip.isr.measureCrosstalk`, and supports the parameters:
+
+        debug.display['reduce'] : `bool`
+            Display a histogram of the combined ratio measurements for
+            a pair of source/target amplifiers from all input
+            exposures/detectors.
+
+        """
+        if outputDims:
+            calibChip = outputDims['detector']
+        else:
+            # calibChip needs to be set manually in Gen2.
+            calibChip = None
+
+        self.log.info("Combining measurements from %d ratios and %d fluxes",
+                      len(inputRatios), len(inputFluxes) if inputFluxes else 0)
+
+        if inputFluxes is None:
+            inputFluxes = [None for exp in inputRatios]
+
+        combinedRatios = defaultdict(lambda: defaultdict(list))
+        combinedFluxes = defaultdict(lambda: defaultdict(list))
+        for ratioDict, fluxDict in zip(inputRatios, inputFluxes):
+            for targetChip in ratioDict:
+                if calibChip and targetChip != calibChip:
+                    raise RuntimeError("Received too many target chips!")
+
+                sourceChip = targetChip
+                if sourceChip in ratioDict[targetChip]:
+                    ratios = ratioDict[targetChip][sourceChip]
+
+                    for targetAmp in ratios:
+                        for sourceAmp in ratios[targetAmp]:
+                            combinedRatios[targetAmp][sourceAmp].extend(ratios[targetAmp][sourceAmp])
+                            if fluxDict:
+                                combinedFluxes[targetAmp][sourceAmp].extend(fluxDict[sourceChip][sourceAmp])
+                # Iterating over all other entries in ratioDict[targetChip] will yield
+                # inter-chip terms.  Skipping to reduce complexity.
```

Comment:
These are currently not populated only because of the middleware issue.  I've updated the comment with the same ticket number used in the Connections class.

Reply:
Brilliant.